scripts/benchmark_all.sh

```diff
@@ -4,23 +4,25 @@ set -e
 
 if [ $# -ne 2 ]
 then
-  echo 'This script is used to compare the performance impact of a change. Running it takes more than an hour.'
+  echo 'This script is used to compare the performance impact of a change. Running it takes a few hours.'
   echo '  It compares two git revisions using various benchmarks (see below) and prints the result in a format'
   echo '  that is copyable to Github.'
   echo 'Typical call (in a release build folder): ../scripts/benchmark_all.sh origin/master HEAD'
   exit 1
 fi
 
 benchmarks='hyriseBenchmarkTPCH hyriseBenchmarkTPCDS hyriseBenchmarkTPCC hyriseBenchmarkJoinOrder'
-warmup_seconds=2
+# Set to 1 because even a single warmup run of a query makes the observed runtimes much more stable. See discussion in #2405 for some preliminary reasoning.
+warmup_seconds=1
+mt_shuffled_runtime=1200
 runs=100
 
-# Setting the number of clients used for the multi-threaded scenario to the machine's core count.
-# This only works for macOS and Linux.
+# Setting the number of clients used for the multi-threaded scenario to the machine's physical core count.
+# This only works for macOS and Linux. For macOS, we simply divide the CPU count by 2.
 output="$(uname -s)"
 case "${output}" in
-    Linux*)     num_phy_cores="$(lscpu -p | egrep -v '^#' | grep '^[0-9]*,[0-9]*,0,0' | wc -l)";;
-    Darwin*)    num_phy_cores="$(sysctl -n hw.ncpu)";;
+    Linux*)     num_phy_cores="$(lscpu -p | egrep -v '^#' | grep '^[0-9]*,[0-9]*,0,0' | sort -u -t, -k 2,4 | wc -l)";;
+    Darwin*)    num_phy_cores="$(($(sysctl -n hw.ncpu) / 2))";;
     *)          echo 'Unsupported operating system. Aborting.' && exit 1;;
```

Comment:
The `sysctl` utility also returns the physical core count. For example:

```
$ sysctl hw | grep cpu
hw.ncpu: 8
hw.activecpu: 8
hw.cpu64bit_capable: 1
hw.cpufamily: 943936839
hw.cpufrequency: 2000000000
hw.cpufrequency_max: 2000000000
hw.cpufrequency_min: 2000000000
hw.cpusubfamily: 0
hw.cpusubtype: 8
hw.cputype: 7
hw.logicalcpu: 8
hw.logicalcpu_max: 8
hw.physicalcpu: 4
hw.physicalcpu_max: 4
hw.cputhreadtype: 1
```

I have a 4-core Intel CPU with hyperthreading support, so `hw.physicalcpu: 4` makes sense. I do not know what the `hw.physicalcpu_max` value should say though.

Reply:
Good point, I will change that to `hw.physicalcpu`

That might be the maximum number of physical CPUs that the mainboard can handle?

Reply:
Yeah, that would make sense.